vllm/utils.py

```diff
@@ -543,11 +543,18 @@ def get_open_zmq_ipc_path() -> str:
     return f"ipc://{base_rpc_path}/{uuid4()}"
 
 
-def get_open_port() -> int:
+def get_open_port(force: bool = False) -> int:
     port = envs.VLLM_PORT
     if port is not None:
+        if force and port is not None:
+            # force vLLM to use envs.VLLM_PORT for torch.distributed init
+            # This is because this port will binded by prefill instance
+            # But both prefill and decode instance need to use this port to
+            # initialize torch.distributed
+            return port
```

Comment:
Should we error out if the port is occupied?

Reply:
In `torch.distributed`, the communication port will be occupied by rank 0. So this port will be occupied by vLLM prefill worker. As a result, in decode worker's view, this port is occupied, but decode worker still need to use this port in order to properly initialize `torch.distributed`.

Reply:
But yeah, this port must be open on prefill worker side. I can add a check that asserts the port must be open on prefill worker side.

Reply:
Fixed in latest commit.